tests/test_components/test_IO.py

```diff
@@ -37,6 +37,8 @@ def set_datasets_to_none(sim):
             src["field_dataset"] = None
         elif src["type"] == "CustomCurrentSource":
             src["current_dataset"] = None
+        if src["source_time"]["type"] == "CustomSourceTime":
```

Comment:
I think we also need to add this to the `Simulation.custom_datasets` property for things to work properly. For example, if not added there, the simulation may not be uploaded properly as an hdf5

https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/simulation.py#L2654-L2670